pkg/obfuscate/sql.go

```diff
@@ -124,7 +207,7 @@ func (f *groupingFilter) Filter(token, lastToken TokenKind, buffer []byte) (toke
 	}
 
 	switch {
-	case f.groupMulti > 0 && lastToken == FilteredGroupableParenthesis && token == ID:
+	case f.groupMulti > 0 && lastToken == FilteredGroupableParenthesis && token == Select:
```

Comment:
Huh, I don't think I follow. Can you explain why only Select is valid here?

Reply:
This change seems unrelated. Also, yes, this whole logic is quite hard to follow 😄

Reply:
Sorry, for not getting around to replying sooner. Before I added the `Select` token [as a keyword](https://github.com/DataDog/datadog-agent/pull/10042/files#diff-c22948a50c7f268987e3e4bc260323ae02b0c5c639538cf2df173aab640d5db0), the SELECT command used to fall under the `ID` token which is essentially like an "other" category that can contain various other SQL values and commands that we have not yet specified. This check here is looking for subqueries to group, and subqueries begin with SELECT. Because I added the `Select` token, the tokenizer is now generating SELECT under the `Select` token, not `ID`, so this check here would never trigger.

Reply:
I see. It makes sense. But couldn't other identifiers/commands start a sub-query? Wouldn't we be playing it safer if we kept both? I mean `token == Select || token == ID` and even `token == Delete`.

Reply:
I did some research before hand regarding SQL grammar and couldn't find anything to suggest that subqueries can be with things like UPDATE or DELETE. However, you're right, we should prepare for the worst regardless and include the "catch all" aspect of this. Sending up a fix, ty @gbbr!